### PR TITLE
Use NixOS on coprocessors.

### PR DESCRIPTION
**Pull request by max-niederman**

### CODE REVIEW

File: coprocessors/flake.nix [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      overlays.default = self: super: {
+        photonvision = self.callPackage ./pkgs/photonvision.nix { };
+        rgb-2024 = rgb-2024.packages.${self.system}.default;
+      };
+
+      devShell = lib.attrsets.genAttrs [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
```
Comment (doinkythederp) [inline coprocessors/flake.nix:77]: FYI I don't think this is working:
```console
$ nix shell
error: flake '...' does not provide attribute 'packages.aarch64-darwin.default' or 'defaultPackage.aarch64-darwin'
```

File: coprocessors/hosts/copepod/hardware-configuration.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+  imports = [ (modulesPath + "/installer/scan/not-detected.nix") ];
+
+  boot.initrd.availableKernelModules = [ "usbhid" ];
+  boot.initrd.kernelModules = [ ];
+  boot.kernelModules = [ ];
+  boot.extraModulePackages = [ ];
+
+  fileSystems."/" = {
+    device = "/dev/disk/by-uuid/44444444-4444-4444-8888-888888888888";
```
Comment (doinkythederp) [inline coprocessors/hosts/copepod/hardware-configuration.nix:15]: this feels like a fake uuid...?

File: coprocessors/configure-linux-router.sh [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+# Enable NAT from the robot subnet to our upstream connection
+echo "Enabling NAT..."
+iptables -t nat -s 10.36.36.0/24 -A POSTROUTING -j MASQUERADE
+
+# Bring the interface up
+echo "Bringing $IFACE up..."
+ip link set $IFACE up
+
+echo "Done configuring $IFACE"
```
[the remark this answers is not in the document]
Comment (max-niederman) [reply coprocessors/configure-linux-router.sh:25]: Wdym outdated? I wrote it myself specifically for this project.

File: coprocessors/flake.nix (hunk 1 above)
Comment (max-niederman) [reply coprocessors/flake.nix:77]: You want `nix develop`.

File: coprocessors/hosts/copepod/hardware-configuration.nix (hunk 2 above)
Comment (max-niederman) [reply coprocessors/hosts/copepod/hardware-configuration.nix:15]: @Gavin-Niederman ?

File: coprocessors/configure-linux-router.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (doinkythederp) [reply coprocessors/configure-linux-router.sh:25]: Well I'm curious if we ever got it working.

File: coprocessors/hosts/copepod/hardware-configuration.nix (hunk 2 above)
Comment (Gavin-Niederman) [reply coprocessors/hosts/copepod/hardware-configuration.nix:15]: Nope this is real. It was generated by nixos-generate-config.

File: coprocessors/flake.nix [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      colmena = {
+        meta = {
+          nixpkgs = import nixpkgs {
+            system = "x86_64-linux";
+            overlays = [ overlays.default ];
+          };
+          specialArgs = { rk3588 = nixos-rk3588.inputs; };
+        };
+
+        copepod = {
```
Comment (doinkythederp) [inline coprocessors/flake.nix:28]: it'd be nice if we could document somewhere what the hostnames actually refer to. e.g. i think the tangerine one is the mini pc

File: coprocessors/hosts/copepod/default.nix [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+  environment.systemPackages = with pkgs; [
+    libraspberrypi
+    raspberrypi-eeprom
+    rgb-2024
+  ];
+
+  nix.settings.trusted-users = [ "root" ];
+
+  system.stateVersion = "23.11"; # Did you read the comment?
```
Comment (doinkythederp) [inline coprocessors/hosts/copepod/default.nix:106]: this one seems to be a newer version from all the others, maybe we should update them? also idk what "the comment" is

**Review by doinkythederp — CHANGES_REQUESTED** [review]

File: coprocessors/flake.nix (hunk 4 above)
Comment (Gavin-Niederman) [reply coprocessors/flake.nix:29]: +1

File: coprocessors/hosts/copepod/default.nix (hunk 5 above)
Comment (Gavin-Niederman) [reply coprocessors/hosts/copepod/default.nix]: Max is working on getting 23.11 working for the other configurations and this is already working. I prefer using more recent state versions when possible.

Comment (Gavin-Niederman) [reply coprocessors/hosts/copepod/default.nix]: I've changed my mind on this and I will set it to 23.05.

**Review by Gavin-Niederman — APPROVED** [review]
LGTM (decLarative Good Tiny Machines)

**Review by doinkythederp — APPROVED** [review]

**Pull request merged**